controller/subscriptionsproxycontroller.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * ownCloud - Calendar App
+ *
+ * @author Georg Ehrke
+ * @copyright 2016 Georg Ehrke <oc.list@georgehrke.com>
+ * @author Raghu Nayyar
+ * @copyright 2016 Raghu Nayyar <beingminimal@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\Calendar\Controller;
+
+use OC\AppFramework\Http;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\DataDisplayResponse;
+use OCP\Files\File;
+use OCP\Files\IRootFolder;
+use OCP\Files\NotFoundException;
+use OCP\IRequest;
+use OCP\IUserSession;
+
+class SubscriptionsProxyController extends Controller {
+
+	/**
+	 * @var IUserSession
+	 */
+	private $userSession;
+
+	/**
+	 * @var \OCP\Files\Folder
+	 */
+	private $storage;
+
+	/**
+	 * @param string $appName
+	 * @param IRequest $request an instance of the request
+	 * @param IUserSession $userSession
+	 * @param IRootFolder $storage
+	 */
+	public function __construct($appName, IRequest $request,
+									IUserSession $userSession, IRootFolder $storage) {
+		parent::__construct($appName, $request);
+		$this->userSession = $userSession;
+		$this->storage = $storage;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @param string $icsurl
+	 * @return DataDisplayResponse
+	 */
+	public function getIcsFile($icsurl) {
+		try {
+			$file = $this->storage->get('/' . basename($icsurl));
```

Comment:
> I'm generally not very happy about storing the webcal data on the server.
This should be really the last resort if we can't circumvent the CSP errors.

Well, I thought of the number of requests it might make for the distant server (especially if https://github.com/owncloud/calendar/issues/599 is active). Also, it's faster. But you can have just a proxy and serve directly the downloaded file.

> Additionally getAppFolder is on a per-app basis, not per-app and per-user.
The current way a user could access all webcal files stored on the server.

Yup, will change.